src/Model/Event.php

```diff
@@ -295,14 +296,18 @@ public static function store(array $arr): int
 				Item::update($fields, ['id' => $item['id']]);
 			}
 
-			Hook::callAll('event_updated', $event['id']);
+			$eventDispatcher->dispatch(
+				new ArrayFilterEvent(ArrayFilterEvent::EVENT_UPDATED, ['event' => $event]),
```

Comment:
shouldn't we differentiate between events with returns and "fire-and-forget" events? So Addons code know that there is no return possible/necessary?

Reply:
Yes, this will be done in a separate step after we've replaced all hooks with events. Then we will have a complete list of all one-way events inside the HookEventBridge and we can start building new events like "fire-and-forget" events.

At the moment an addon developer has to check if the (referenced) argument to `Hook::callAll()` will be used later. Now one can see that the returned event of `$eventDispatcher->dispatch()` is not used.